src/libsyntax/tokenstream.rs

```diff
@@ -532,12 +685,14 @@ impl TokenSlice {
     }
 
     /// Returns the RHS of an assigment.
-    pub fn maybe_assignment(&self) -> Option<&TokenSlice> {
+    pub fn maybe_assignment(&self) -> Option<TokenStream> {
```

Comment:
I'm a bit sceptical of including stuff because it *might* be useful for macro writers. Libmacro should be as minimal as possible with stuff like this in external libraries. We should include stuff only when it is impossible to have it outside or that something outside has proved itself to be absolutely essential.